### PR TITLE
Add dns manual verification

### PR DESCRIPTION
**Pull request by mjmayer**
- fixes rst formatting errors in `contact_and_support.rst` and `slack_webhook.rst`
- Adds a page on manual verification of MX, DKIM, and SPF records
- Add custom anchor to the email hosting page to link to from the `dns_verifications.rst` page

The dns verification page can be review at RTD https://docs.mailsac.com/en/add_dns_manual_verification/help/dns_verification/dns_verification.html

### CODE REVIEW

**Comment by ruffrey** [comment]
image

I think this warrants more discussion. First potentially confusing thing here is the msdc.co subdomain, which customers won't ever need to configure on their own.
Second potentially confusing thing is it shows the full subdomain path for the DNS entry. However for most hosts, you only enter the subdomain or @.

File: help/dns_verification/dns_verification.rst [hunk 1]
```diff
@@ -0,0 +1,87 @@
+.. _doc_dnsverification:
+.. _Dashboard: https://mailsac.com/dashboard
+.. _Domains: https://mailsac.com/domains
+
+DNS Verification
+================
+
+DNS (Domain Name System) is at the core of the internet and email delivery.
+When configuring a :ref:`BYO Domain <section_byod_configuration>` the following
+records need to be configured:
```
Comment (ruffrey) [inline help/dns_verification/dns_verification.rst:10]: The MX records are optional, so not "need to be configured"

The verification of domain ownership only requires the DKIM and SPF records.

File: help/dns_verification/dns_verification.rst [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+MX Record Verification
+----------------------
+
+The values of the MX Records are the :ref:`same <section_email_hosting>` for all
+Private and Public Domains hosted by Mailsac. The Host will be unique per
+domain.
+
+======== ================= ==================
+**Type** **Host**           **Value**
```
Comment (ruffrey) [inline help/dns_verification/dns_verification.rst:24]: If we're walking people all the way through these records, explaining what the `1` and `5` mean would be a good idea. I have seen registrars where they have you fill out a little form indicating the priority number, with the MX hostname separate.

File: help/dns_verification/dns_verification.rst [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+DKIM Record Verification
+------------------------
+
+DKIM Records are unique per domain. The value of the required DKIM record can
+be found in the Dashboard_ under Domains_, then click "Manage" and choose the
+"DNS Setup" tab.
+
+======== ========================================== ========================
+**Type** **Host**                                   **Value**
+TXT      mailsacrelay._domainkey.dogtreats.msdc.co. v=DKIM1; k=rsa; p=MI....
```
Comment (ruffrey) [inline help/dns_verification/dns_verification.rst:50]: A user contacted me about a week and a half ago. He had entered exactly the host like this. But in his case, his domain provider (like most) does not want the `dogtreats.msdc.co` part, only `mailsacrelay._domainkey.dogtreats`. It took a lot of back and forth, and me guessing to run dig against effectively `dig mailsacrelay._domainkey.dogtreats.msdc.co.msdc.co` (yes) to even determine this issue.

Perhaps we need to call out that this should be the "end result" of entering the records. Showing examples of:

- configuring example.com
- configuring mail.example.com
- configuring example.co.uk
- configuring mail.example.co.uk

**Comment by mjmayer** [comment]
> Unfortunately this may further confuse people. We require DKIM and SPF to verify their domain. These are also required for sending mail.
> 
> Someone who only wants to receive mail - nearly all our customers - will read this as "I only need MX records added below to verify my domain." But it won't work like that.

I've change the wording to match the wording that is on the "DNS Setup" page in the frontend application.

**Pull request merged**